### PR TITLE
Optimization

### PR DESCRIPTION
**Pull request by babenek**
## Description

- Added limits for data which might be represented for deep scan
- Fix UTF-8 surrogates issue during encoding

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest

### CODE REVIEW

File: tests/samples/sample.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
+#!/bin/env python
 def get_password() -> str:
     password = \
         "WeR15tr0n6"
     return password
 
 
-if "__main__" == __name__:
+TEST = "\uDF42\uD824"
```
Comment (csh519) [inline tests/samples/sample.py:8]: Why this `TEST` variable added?

Comment (babenek) [reply tests/samples/sample.py:8]: The file is a sample for test. 
When the sample is parsed with python ast with using '--depth' - it fails with encoding.
E.g.: 
```
>>> print ("\uDF42\uD824")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 0-1: surrogates not allowed
```
So, i've added the sample to trigger test fail in future. Like a canary test.

Comment (csh519) [reply tests/samples/sample.py:8]: I see. Thank you for explanation.

**Review by csh519 — APPROVED** [review]
Code is optimized and UTF-8 issue is fixed.

LGTM 👍

**Review by iuriimet — APPROVED** [review]

**Pull request merged**